### PR TITLE
Make `_magicgui.pyi` stubs

### PR DESCRIPTION
**Pull request by tlambert03**
The type annotations in `_magicgui.py` got a little unsightly in #117.  This puts all of the overloads and other typing stuff into `_magicgui.pyi`.  Tests for typing annotations were added (in the `typesafety` folder) in #117 and are still passing here.

### CODE REVIEW

File: magicgui/_magicgui.pyi [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+    labels: bool = True,
+    tooltips: bool = True,
+    call_button: Union[bool, str] = False,
+    auto_call: bool = False,
+    result_widget: bool = False,
+    main_window: Literal[True],
+    app: AppRef = None,
+    **param_options: dict,
+) -> Callable[[Callable[..., _R]], MainFunctionGui[_R]]: ...
+@overload  # noqa: E302
```
Comment (jni) [inline magicgui/_magicgui.pyi:80]: Why are you squishing this so much?

**Review by jni — APPROVED** [review]
Wow, you figured it out??? What was the issue?

File: magicgui/_magicgui.pyi (hunk 1 above)
Comment (tlambert03) [reply magicgui/_magicgui.pyi:80]: that's what black does to pyi files! :joy:

Comment (tlambert03) [reply magicgui/_magicgui.pyi:80]: > What was the issue?

turns out it was just me being dumb.  I was trying to test it interactively _in_ the `_magicgui.py` file.  Now that I have the tests in `typesafety` I realized the were working even though it didn't seem to be working inside of `_magicgui.py` ... so I just tried somewhere else and it worked!

Comment (jni) [reply magicgui/_magicgui.pyi:80]: > that's what black does to pyi files! 😂

🤢

**Pull request merged**